### PR TITLE
AtlasEngine: Harden against invalid soft fonts

### PR DESCRIPTION
**Pull request by lhecker**
This is a theoretical improvement for #15553 where Windows Terminal
crashed due to AtlasEngine accessing the soft font bitmap outside of
bounds. The problem is that the soft font cell size was non-zero.
This PR hardens against such situations by checking whether the
requested soft font index is inside the bounds of the bitmaps.
The improvement couldn't be tested as it couldn't be reproduced.

### CODE REVIEW

**Review by DHowett — APPROVED** [review]

File: src/renderer/atlas/AtlasEngine.cpp [hunk 1]
```diff
@@ -631,13 +631,12 @@ void AtlasEngine::_flushBufferLine()
```
Comment (captain-hoity-toity) [inline src/renderer/atlas/AtlasEngine.cpp:631]: A few more unit tests might help cover edge cases.

Comment (lhecker) [reply src/renderer/atlas/AtlasEngine.cpp:631]: I agree with you, however in this particular instance you'd have to tell me what the edge case is that causes the issue, because no one knows right now and it's not possible to write a test for something unknown.

**Comment by lhecker** [comment]
I just realized you're using GPT generated content. Not appreciated. 🙂

**Review by zadjii-msft — APPROVED** [review]
